File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/progresskeeper/ProgressListener.java
package net.kdt.pojavlaunch.progresskeeper;

/**
 * 進行状況の変更をリッスンするインターフェース。
 */
public interface ProgressListener {
    /** 進行状況が開始されたときに呼び出されます。 */
    void onProgressStarted();
    /** 進行状況が更新されたときに呼び出されます。 */
    void onProgressUpdated(int progress, int resid, Object... va);
    /** 進行状況が終了したときに呼び出されます。 */
    void onProgressEnded();
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/task/ProgressListener.kt
package com.arata.yukarilauncher.task

/** 進捗状態の変更を監視するリスナー。 */
interface ProgressListener {
    /** 進捗が開始されたときに呼び出される。 */
    fun onProgressStarted()
    /** 進捗が更新されたときに呼び出される。 @param progress 進捗値 @param resid リソースID @param va 追加引数 */
    fun onProgressUpdated(progress: Int, resid: Int, vararg va: Any?)
    /** 進捗が終了したときに呼び出される。 */
    fun onProgressEnded()
}
